misaligned.c

```diff
@@ -1,13 +1,30 @@
 #include <stdio.h>
 #include <assert.h>
+#include <string.h>
+#include "misaligned_config.h"
+
+#ifdef UNIT_TEST

```

Comment:
Using conditional compilation for separating production and test code may get complicated as the code size grows.
it is better to use interfaces or functions pointers or lambda functions (if available) to inject either production or test code based on the envt.
Production code should be simple in the sense, given an input the output should be deterministic, may it be dev environment or test or field